app/controllers/newflow/login_controller.rb

```diff
@@ -3,17 +3,14 @@ class LoginController < BaseController
 
     include LoginSignupHelper
 
-    GO_TO_STUDENT_SIGNUP = 'student_signup'
-    GO_TO_SIGNUP = 'signup'
-
     fine_print_skip :general_terms_of_use, :privacy_policy, except: :profile_newflow
 
     before_action :cache_client_app, only: :login_form
     before_action :known_signup_role_redirect, only: :login_form
     before_action :cache_alternate_signup_url, only: :login_form
     before_action :redirect_to_signup_if_go_param_present, only: :login_form
-    before_action :did_sign_privacy_notice, if: -> { signed_in? }, only: :login_form
-    before_action :redirect_back, if: -> { signed_in? }, only: :login_form
+    before_action :redirect_to_sign_privacy_notice, if: -> { signed_in? && !did_user_sign_recent_privacy_notice? }, only: :login
```

Comment:
Shouldn't this be `login_form`? After this change there's no redirect for `login_form` if they are signed in but did not sign the notice

Reply:
It seemed to work for me with `:login`, wasn't sure what the difference is but both seem to work.

Does this resolve your second point about being no redirect for `login_form` if signed in but didn't sign?

Reply:
does this still work as you expect? Otherwise you may need both

Reply:
I see what you mean, yes, I do need both of them for it to work correctly.